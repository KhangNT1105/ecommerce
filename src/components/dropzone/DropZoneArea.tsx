import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useDropzone } from 'react-dropzone';
import { clone } from 'lodash';
import get from 'lodash/get';

import { IDropZone } from './DropZoneAreaType';
import DropzonePreview from './preview/DropzonePreview';
import API from 'api/';
import {
  UPLOAD_ACCEPT_FILE_TYPE,
  FILE_DROPZONE_MIN_SIZE,
  FILE_DROPZONE_MAX_SIZE,
  RESOLUTION_MIN_SIZE_LANDSCAPE,
  RESOLUTION_MIN_SIZE_PORTRAIT
} from 'constants/dropzone';
import { DROPZONE_EVENT_TYPE } from 'constants/enum';
import { notify } from 'components/toast/Toast';

import 'components/dropzone/Dropzone.scss';

const DropZoneArea: React.FC<IDropZone> = (props) => {
  const { t } = useTranslation();
  const { apiURL, files, autoUpload, config, getData, pushData } = props;

  const [fileHandle, setfileHandle] = useState<File[]>([]);
  const [filesUpload, setFilesUpload] = useState<boolean>(false);
  const [fileResponse, setFileResponse] = useState<object>();
  const [isDrag, setIsDrag] = useState<boolean>(false);
  const [resolutionMediaMinSize, setResolutionMediaMinSize] = useState<string>('');
  const [mediaTypeError, setMediaTypeError] = useState<boolean>(false);

  const form = React.useRef() as React.MutableRefObject<HTMLFormElement>;

  const formData = new FormData(form.current);

  let eventStatus: string;
  const onDrop = (files: File[], rejectedFiles: File[]) => {
    if (!rejectedFiles) {
      pushData.infos.fileName = files[0].name;
    }

    setfileHandle(files);
    // Check bypass for autoupload
    if (autoUpload || !autoUpload || eventStatus !== DROPZONE_EVENT_TYPE.DROP) {
      setIsDrag(true);
      onFileUpload(files);
    }

    // check image dimension
    // clone file to avoid control param
    const dropFiles = (rejectedFiles.length > 0 && clone(rejectedFiles)) || clone(files);
    if (config.accept.indexOf('image/') > -1) {
      // call FileReader to get media's solution
      const reader = new FileReader();
      reader.onload = function () {
        const image: any = new Image();
        image.src = reader.result;

        image.onload = function () {
          let resolutionMedia = '';
          // validate if image dimension is smaller than requirement
          if (image.width < RESOLUTION_MIN_SIZE_LANDSCAPE || image.height < RESOLUTION_MIN_SIZE_PORTRAIT) {
            // is landscape
            resolutionMedia = `${RESOLUTION_MIN_SIZE_LANDSCAPE}x${RESOLUTION_MIN_SIZE_PORTRAIT}`;
            // is portrait
            if (image.width < image.height) {
              resolutionMedia = `${RESOLUTION_MIN_SIZE_PORTRAIT}x${RESOLUTION_MIN_SIZE_LANDSCAPE}`;
            }
          } else {
            resolutionMedia = '';
          }

          // set resolution media
          setResolutionMediaMinSize(resolutionMedia);
        };
      };

      // call readAsDataURL to recall onload fn
      reader.readAsDataURL(dropFiles[0]);
    }
  };

  const onDropRejected = (rejectedFiles: File[]) => {
    if (rejectedFiles.length > 0) {
      // validate if image type is not in requirement
      if (props.config.accept.indexOf(rejectedFiles[0].type) === -1) {
        setMediaTypeError(true);
      }
    }
  };

  const { rejectedFiles, getRootProps, getInputProps, isDragActive, isDragReject } = useDropzone({
    ...config,
    onDrop,
    onDropRejected
  });

  const isFileTooLarge = rejectedFiles.length > 0 && rejectedFiles[0].size > config.maxSize;

  // Remove a file
  const onRemoveFile = (file: number) => {
    setfileHandle(files.slice(file));

    // reset all prev state
    setMediaTypeError(false);
    setResolutionMediaMinSize('');
  };

  const onFileUpload = async (files: File[]) => {
    if (files.length > 0) {
      files.forEach((file: File) => {
        formData.append('files', file, file.name);
      });
      await postData(formData);
    }
  };

  const onSendDatas = (datas: object) => {
    if (getData) {
      getData && getData(datas);
    }
  };

  const postData = async (datas: any) => {
    try {
      await API({
        url: apiURL,
        params: '',
        method: 'POST',
        headers: { 'Content-Type': 'multipart/form-data' },
        data: datas
      })
        .then((resp: any) => {
          notify.success(t('CREATED_CAST_SUCCESSFULLY'));
          if (resp) {
            setFilesUpload(true);
            setFileResponse(pushData);
            onSendDatas(pushData);
          }
        })
        .catch((error: any) => {
          const userMessage = get(error, 'response.userMessage', '');
          if (userMessage) {
            notify.error(userMessage);
            return;
          }
        });
    } catch {
      setFilesUpload(false);
    }
  };

  const onDragHandle = (e: React.MouseEvent) => {
    eventStatus = e.type;
  };

  const renderErrorMessage = () => {
    const mimeType = config.accept.replace(new RegExp('image/|video/', 'g'), '').toUpperCase();

    return (
      <div className="error-container">
        <small className="small">
          {!isDragActive && !isFileTooLarge && !resolutionMediaMinSize && t('DROPZONE_IS_DRAG_ACTIVE')}
        </small>
        <small className="small">{isDragActive && !isDragReject && t('DROPZONE_DROP_TO_FILE')}</small>
        <small className="small">{mediaTypeError && t('DROPZONE_FILE_TYPE_NOT_ALLOWED', { mimeType })}</small>
        <small className="small">
          {isFileTooLarge && t('DROPZONE_IS_FILE_TOO_LARGE', { maxSize: config.maxSize })}
        </small>
        <small className="small">
          {resolutionMediaMinSize &&
            t('DROPZONE_IS_FILE_RESOLUTION_TOO_SMALL', {
              resolutionMediaMinSize
            })}
        </small>
      </div>
    );
  };

  const renderReview = () => {
    if (!isFileTooLarge && !resolutionMediaMinSize) {
      return (
        <DropzonePreview
          files={fileHandle}
          onFileRemove={onRemoveFile}
          onFileUpload={() => onFileUpload(fileHandle)}
          status={filesUpload}
          datas={fileResponse}
          drag={isDrag}
        />
      );
    }
  };

  return (
    <section className="dropzone--wrapper">
      <div
        {...getRootProps({
          className: 'dropzone',
          onDrop: (e) => {
            onDragHandle(e);
          },
          onClick: (e) => {
            onDragHandle(e);
          }
        })}
        className="dropzone"
      >
        <div>
          <i className="fa fa-cloud-upload fa-fw fa-3x mb-3" />
          <form encType="multipart/form-data" ref={form}>
            <input {...getInputProps()} />
          </form>
          <h5 className="mt-0">{t('DROPZONE_MENTION_TEXT')}</h5>

          {renderErrorMessage()}
        </div>
      </div>
      {renderReview()}
    </section>
  );
};

DropZoneArea.defaultProps = {
  apiURL: '',
  files: [],
  config: {
    accept: UPLOAD_ACCEPT_FILE_TYPE,
    disabled: false,
    noClick: false,
    noKeyboard: false,
    minSize: FILE_DROPZONE_MIN_SIZE,
    maxSize: FILE_DROPZONE_MAX_SIZE /* 1 MB */
  }
};

export default DropZoneArea;
