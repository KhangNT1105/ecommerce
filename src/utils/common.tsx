import React from 'react';
import { Denomination } from 'Props.d';
import { assign, clone, map, reduce, isObjectLike } from 'lodash';
import { validateRequired, validateEmail } from 'utils/validation';
import { ACTIVATE, ENTITY_STATUS_FROM_SERVER, DEACTIVATE, DELETE, APPROVE, REJECT } from '../constants';
import { Genre, SubGenre } from 'stores/ConfigurationStore/configurationType.d';

export const checkRenderDoubleValue = (value1?: string | number, value2?: string | number, value3?: any | '') => {
  let minus = '';
  if (value1 && value2) {
    minus = ' - ';
  }
  return (
    <>
      {value1} {value3 && <b>{value3}</b>}
      {minus}
      {value2}
    </>
  );
};

export type SelectOptionProps = {
  id: string;
  value: string;
  label: string;
};

type DataProps = {
  [key: string]: any;
};

export type SelectOptionKeyMappingProps = {
  key: string;
  value: string;
  label: string;
};

export const renderSelectOptions = (data: DataProps[], keyMapping?: SelectOptionKeyMappingProps) => {
  return (
    <>
      <option value="">Select</option>
      {data.map((item) => {
        if (keyMapping && Object.entries(keyMapping).length > 0) {
          return (
            <option key={item[keyMapping.key]} value={item[keyMapping.value]}>
              {item[keyMapping.label]}
            </option>
          );
        }
        return (
          <option key={item.id} value={item.value}>
            {item.label}
          </option>
        );
      })}
    </>
  );
};

export const checkRenderArray = (items: string[]) => {
  return <span>{items.join(', ')}</span>;
};

export const getListPrices = (denominations: Denomination[]) => {
  const listPrices = denominations.map((item) => item.price);
  return listPrices.sort((a, b) => a - b);
};
export const getPrices = (productDetail: any) => {
  if (productDetail.valueType === 'RANGE') {
    return checkRenderDoubleValue(productDetail.minPrice, productDetail.maxPrice);
  }
  if (productDetail.valueType === 'SLAB') {
    const values = getListPrices(productDetail.denominations);
    return values.join(', ');
  }
};

export interface IPagingModelOption {
  pageIndex: number;
  pageSize: number;
}

export interface IPagingModelResult<T> {
  items: T[];
  hasPrevPage: boolean;
  hasNextPage: boolean;
  prevPage: number | null;
  nextPage: number | null;
  size: number;
  page: number;
  totalItems: number;
  totalPages: number;
}

export function getPaginationResponse<T>(
  pagingModel: IPagingModelOption,
  result: T[],
  totalItems: number
): IPagingModelResult<T> {
  const totalPages = Math.floor((totalItems + pagingModel.pageSize - 1) / pagingModel.pageSize);
  const pagingModelResult: IPagingModelResult<T> = {
    items: result,
    totalItems,
    totalPages,
    hasPrevPage: pagingModel.pageIndex > 0,
    hasNextPage: pagingModel.pageIndex < totalPages - 1,
    prevPage: pagingModel.pageIndex > 0 ? pagingModel.pageIndex - 1 : null,
    nextPage: pagingModel.pageIndex < totalPages - 1 ? pagingModel.pageIndex + 1 : null,
    size: pagingModel.pageSize,
    page: pagingModel.pageIndex
  };
  return pagingModelResult;
}

export function replaceSpecialCharacters(str: string, character: string) {
  return str.replace(/(?:\.(?![^.]+$)|[^\w.])+/g, character);
}

export const classConcat = (classes: string[]): string => {
  if (classes.length) {
    return classes.filter((className) => !!className).join(' ');
  }
  return '';
};

export const recursiveGetFieldName = (obj: object, entityField: string, fieldNameCombine?: string): any => {
  let listFieldConfig: any;
  listFieldConfig = reduce(
    obj,
    (result, value, key) => {
      let cloneResult = clone(result);
      let localFieldName: any = clone(fieldNameCombine);
      if (key === entityField) {
        map(value, (item: any, k) => {
          const getFieldName = `${localFieldName}.${key}.${k}`;

          // check whether any validation
          if (item.validationRules && item.validationRules.length > 0) {
            item.validationRules.map((itemValidate: any, index: number) => {
              if (itemValidate.rule === 'isRequired') {
                item.validationRules[index].validator = validateRequired;
              }
              if (itemValidate.rule === 'isEmail') {
                item.validationRules[index].validator = validateEmail;
              }

              return itemValidate;
            });
          }

          // insert key, value into list field
          return (cloneResult = {
            ...cloneResult,
            [getFieldName]: item
          });
        }); // reset field name

        localFieldName = '';
      } else if (isObjectLike(value)) {
        if (!localFieldName) {
          localFieldName = key;
        } else {
          localFieldName += `.${key}`;
        } // recall recursive fn if not be found the expected entityField

        return assign(cloneResult, recursiveGetFieldName(value, entityField, localFieldName));
      }

      listFieldConfig = {
        ...listFieldConfig,
        ...cloneResult
      };

      return listFieldConfig;
    },
    {}
  );

  return listFieldConfig;
};

export type ObjectFieldType = {
  [field: string]: string | object | string[] | object[] | any;
};

export const trimAllValueWithObjectField = (objField: ObjectFieldType) => {
  if ((!Array.isArray(objField) && typeof objField !== 'object') || objField === null) {
    return objField;
  }
  return Object.keys(objField).reduce(
    (acc: ObjectFieldType, key) => {
      acc[key] =
        typeof objField[key] === 'string'
          ? (objField[key] as string).trim()
          : trimAllValueWithObjectField(objField[key]);
      return acc;
    },
    Array.isArray(objField) ? [] : {}
  );
};

const listShowBtnActivate = [
  ENTITY_STATUS_FROM_SERVER.INACTIVATE,
  ENTITY_STATUS_FROM_SERVER.DRAFT,
  ENTITY_STATUS_FROM_SERVER.UPDATED,
  ENTITY_STATUS_FROM_SERVER.ACTIVATE
];

const listShowBtnSMActivate = [
  ENTITY_STATUS_FROM_SERVER.INACTIVATE,
  ENTITY_STATUS_FROM_SERVER.DRAFT,
  ENTITY_STATUS_FROM_SERVER.UPDATED
];

const listShowBtnSMDelete = [ENTITY_STATUS_FROM_SERVER.DRAFT, ENTITY_STATUS_FROM_SERVER.INACTIVATE];

const listShowBtnDeactivate = [ENTITY_STATUS_FROM_SERVER.ACTIVATE, ENTITY_STATUS_FROM_SERVER.UPDATED];

const listShowBtnDelete = [ENTITY_STATUS_FROM_SERVER.INACTIVATE, ENTITY_STATUS_FROM_SERVER.DRAFT];

export const checkEntityActionNonApproval = (action: string, entityStatus: string) => {
  switch (action) {
    case ACTIVATE:
      return listShowBtnActivate.includes(entityStatus);
    case DEACTIVATE:
      return listShowBtnDeactivate.includes(entityStatus);
    case DELETE:
      return listShowBtnDelete.includes(entityStatus);
    default:
      return true;
  }
};

export const checkSMEntityActionNonApproval = (action: string, entityStatus: string) => {
  switch (action) {
    case ACTIVATE:
      return listShowBtnSMActivate.includes(entityStatus);
    case DEACTIVATE:
      return listShowBtnDeactivate.includes(entityStatus);
    case DELETE:
      return listShowBtnSMDelete.includes(entityStatus);
    default:
      return true;
  }
};

export const getAllSubGenres = (genres: Genre[]) =>
  genres
    .map(({ children }) => children)
    .reduce((currentValue, subGenres) => {
      if (subGenres) {
        return [...currentValue, ...subGenres];
      }
      return currentValue;
    }, []);

export const getAllMicroGenres = (subGenres: SubGenre[]) =>
  subGenres
    .map(({ children }) => children)
    .reduce((currentValue, microGenre) => {
      if (microGenre) {
        return [...currentValue, ...microGenre];
      }
      return currentValue;
    }, []);

const listShowBtnActivateWithApproval = [...listShowBtnActivate, ENTITY_STATUS_FROM_SERVER.REJECT];
const listShowBtnDeactivateWithApproval = [...listShowBtnDeactivate];
const listShowBtnDeleteWithApproval = [...listShowBtnDelete];
const listShowBtnApproveWithApproval = [ENTITY_STATUS_FROM_SERVER.APPROVAL_PENDING];
const listShowBtnRejectWithApproval = [ENTITY_STATUS_FROM_SERVER.APPROVAL_PENDING];

export const checkEntityActionApproval = (action: string, entityStatus: string) => {
  switch (action) {
    case ACTIVATE:
      return listShowBtnActivateWithApproval.includes(entityStatus);
    case DEACTIVATE:
      return listShowBtnDeactivateWithApproval.includes(entityStatus);
    case DELETE:
      return listShowBtnDeleteWithApproval.includes(entityStatus);
    case APPROVE:
      return listShowBtnApproveWithApproval.includes(entityStatus);
    case REJECT:
      return listShowBtnRejectWithApproval.includes(entityStatus);
    default:
      return true;
  }
};

export function createMarkup() {
  return {
    __html: `<div class="zalo-chat-widget" data-oaid="927003371386666481" data-welcome-message="Rất vui khi được hỗ trợ bạn!" data-autopopup="0" data-width="450" data-height="450"></div>
    `
  };
}
