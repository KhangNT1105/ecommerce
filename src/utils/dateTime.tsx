import {
  format,
  parse,
  getTime,
  subDays,
  subMonths,
  startOfWeek,
  endOfWeek,
  startOfMonth,
  endOfMonth,
  startOfToday,
  endOfToday,
  startOfDay,
  endOfDay
} from 'date-fns';

import { DATE_FORMAT, DATE_TIME_FULL_FORMAT } from 'constants/index';

export type DateRangeType = { from: string | null; to: string | null };

export const formatDate = (timestamp: string) => {
  return format(parse(timestamp, 'T', new Date()), DATE_FORMAT);
};

export type DateRangeProps = {
  dateFrom: string | null;
  dateTo: string | null;
};

/**
 * get timestamp from a string
 * @param: value string
 */
export const getTimestamp = (value: string): number => {
  return getTime(parse(value, DATE_FORMAT, new Date()));
};

export const getUTCDateFn = (date: Date) => {
  return new Date(
    date.getUTCFullYear(),
    date.getUTCMonth(),
    date.getUTCDate(),
    date.getUTCHours(),
    date.getUTCMinutes(),
    date.getUTCSeconds()
  );
};

/**
 * Description: combine day from DatePicker and local time then convert to UTC Date
 * @param date: from date picker
 * @return: UTC Date
 */
export const convertDatePickerToUTC = (date: Date) => {
  const today = new Date();
  const combineDateTime = new Date(
    date.getFullYear(),
    date.getMonth(),
    date.getDate(),
    today.getHours(),
    today.getMinutes(),
    today.getSeconds()
  );

  return getUTCDateFn(combineDateTime);
};

/**
 * Description: convert HH:MM:SS string to seconds
 * @param dateString
 * @param second
 */
export const convertTimeStringToMillisecond = (dateString: String) => {
  return (dateString as any).split(':').reduce((acc: any, time: any) => 60 * acc + +time) * 1000;
};

export const convertDateFromToUTC = (date: Date) => {
  const combineDateTime = new Date(date.getFullYear(), date.getMonth(), date.getDate());

  return getUTCDateFn(startOfDay(combineDateTime));
};

export const convertDateToToUTC = (date: Date) => {
  const combineDateTime = new Date(date.getFullYear(), date.getMonth(), date.getDate());

  return getUTCDateFn(endOfDay(combineDateTime));
};

export const getToday = (): DateRangeProps => {
  const dateFrom = getUTCDateFn(startOfToday());
  const dateTo = getUTCDateFn(endOfToday());

  return {
    dateFrom: format(dateFrom, DATE_TIME_FULL_FORMAT),
    dateTo: format(dateTo, DATE_TIME_FULL_FORMAT)
  };
};

export const getLast7days = (): DateRangeProps => {
  const dateTo = format(getUTCDateFn(endOfToday()), DATE_TIME_FULL_FORMAT);
  const dateFrom = format(getUTCDateFn(subDays(startOfToday(), 7)), DATE_TIME_FULL_FORMAT);

  return { dateFrom, dateTo };
};

export const getLast30days = (): DateRangeProps => {
  const dateTo = format(getUTCDateFn(endOfToday()), DATE_TIME_FULL_FORMAT);
  const dateFrom = format(getUTCDateFn(subDays(startOfToday(), 30)), DATE_TIME_FULL_FORMAT);

  return { dateFrom, dateTo };
};

export const getLastWeek = (): DateRangeProps => {
  const dateFrom = format(getUTCDateFn(startOfWeek(subDays(startOfToday(), 7))), DATE_TIME_FULL_FORMAT);
  const dateTo = format(getUTCDateFn(endOfWeek(subDays(endOfToday(), 7))), DATE_TIME_FULL_FORMAT);

  return { dateFrom, dateTo };
};

export const getLastMonth = (): DateRangeProps => {
  const dateFrom = format(getUTCDateFn(startOfMonth(subMonths(startOfToday(), 1))), DATE_TIME_FULL_FORMAT);
  const dateTo = format(getUTCDateFn(endOfMonth(subMonths(endOfToday(), 1))), DATE_TIME_FULL_FORMAT);

  return { dateFrom, dateTo };
};

export const convertMillisecondToHHMMSS = (milliseconds: number) => {
  if (!milliseconds) return '';
  const date = new Date(milliseconds);
  const hh = date.getUTCHours();
  const mm = date.getUTCMinutes();
  const ss = date.getSeconds();
  const hours = hh < 10 ? `0${hh}` : `${hh}`;
  const minutes = mm < 10 ? `0${mm}` : `${mm}`;
  const seconds = ss < 10 ? `0${ss}` : `${ss}`;

  return `${hours}:${minutes}:${seconds}`;
};
