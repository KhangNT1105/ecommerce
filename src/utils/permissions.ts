import { matchPath } from 'react-router';
import { FieldType, PermissionFieldsType, FormPermissionsType } from '../components/roleFormField/RoleFormFields.d';
import { BUTTONS_VISIBILITY, PERMISSION_PREFIX, PAGE_ACCESSIBILITY } from '../constants/enum';
/**
 * Description: check user permissions have spec permissions
 * @param permissions: array of user permissions
 * @param modes: array of PERMISSION_PREFIX
 * Results: boolean
 */
export const hasModes = (permissions: string[], modes: string[]) => {
  const foundMode = permissions.filter((item) => {
    const includes = modes.filter((mode) => {
      return item === mode;
    });
    return includes.length > 0;
  });
  return foundMode.length === modes.length;
};

/**
 * Description: check user permissions only have 1 permission
 * @param permissions: array of user permissions
 * @param mode: PERMISSION_PREFIX
 * Results: boolean
 */
export const hasOnlyMode = (permissions: string[], mode: string) => {
  const viewMode = permissions.filter((item) => {
    return item === mode;
  });
  return permissions.length === 1 && viewMode.length === 1;
};

/**
 * Description: generate new structure data:
 * Params:
 *    @param formPermissions: array of permissions and element name (id)
 *    @param userPermissions: array of permissions of logged in users
 *
 * Results:
 * {
 *  btnSave: 'show',
 *  btnCancel: 'NA',
 *  ...
 * }
 */
export const reGeneratePermissionFields = (formPermissions: FormPermissionsType[], userPermissions: string[]) => {
  const permissionFields: PermissionFieldsType = formPermissions
    .filter(({ permissions }) => {
      return userPermissions.includes(permissions);
    })
    .map((formPermissions: FormPermissionsType) => {
      const { permissions, ...restPermissions } = formPermissions;
      return restPermissions;
    });

  const fieldsWithPermissions = permissionFields.reduce(
    (accumulator: FieldType, currentValue: FieldType, index: number) => {
      const fields = {
        ...accumulator
      };
      Object.keys(currentValue).map((operation: string) => {
        if (currentValue[operation] === BUTTONS_VISIBILITY.SHOW) {
          fields[operation] = currentValue[operation];
        }
        return true;
      });
      return {
        ...fields
      };
    },
    {}
  );
  return fieldsWithPermissions;
};
/**
 * Description: get accessibility of page with specification page
 * Params:
 *    @param formPermissions: array of permissions of page
 *    @param userPermissions: array of permissions of logged in users
 *    @param strPermission: permission
 *
 * Results:
 * pageAccessibility: boolean
 */
export const checkPageAccessibility = (
  formPermissions: FormPermissionsType[],
  userPermissions: string[],
  strPermission: string
) => {
  const pagePermissions = reGeneratePermissionFields(formPermissions, userPermissions);
  return pagePermissions[strPermission] === PAGE_ACCESSIBILITY.SHOW;
};

/**
 * Description: get accessibility of page with specification entity
 * Params:
 *    @param path: current path of site
 *    @param entityName: entity
 *    @param formPermissionsCreate: array of permissions of create page
 *    @param formPermissionsEdit: array of permissions of edit page
 *    @param formPermissionsList: array of permissions of list page
 *    @param pathCreate: route string of create page
 *    @param pathEdit: route string of edit page
 *    @param userPermissions: array of permissions of logged in users
 *
 * Results:
 * pageAccessibility: boolean
 */
export const getPageAccessibility = (
  path: string,
  entityName: string,
  formPermissionsCreate: FormPermissionsType[],
  formPermissionsEdit: FormPermissionsType[],
  formPermissionsList: FormPermissionsType[],
  pathCreate: string,
  pathEdit: string,
  userPermissions: string[]
) => {
  let pageAccessibility = true;
  if (matchPath(path, { path: pathCreate, exact: true })) {
    pageAccessibility = checkPageAccessibility(
      formPermissionsCreate,
      userPermissions,
      `${entityName}_${PERMISSION_PREFIX.CREATE.toLowerCase()}`
    );
  } else if (matchPath(path, { path: pathEdit, exact: true })) {
    pageAccessibility = checkPageAccessibility(
      formPermissionsEdit,
      userPermissions,
      `${entityName}_${PERMISSION_PREFIX.EDIT.toLowerCase()}`
    );
  } else {
    pageAccessibility = checkPageAccessibility(formPermissionsList, userPermissions, `${entityName}`);
  }
  return pageAccessibility;
};

/**
 * Description: check page is accessible by entity
 * Params:
 *    @param page: page name
 *    @param path: current path of site
 *    @param userPermissions: array of permissions of logged in users
 *
 * Results:
 * pageAccessibility: boolean
 */
export const checkPagePermissisons = (page: string, path: string, userPermissions: string[]) => {
  // let pagePermissions: { [permission: string]: string };
  const pageAccessibility = true;

  return pageAccessibility;
};

/**
 * Description: get form permissions of list page by entity
 * Params:
 *    @param entity: entity name
 *
 * Results:
 * formPermissions: FormPermissionsType[]
 */
export const getFormPermissionsPageList = (entity: string) => {};
