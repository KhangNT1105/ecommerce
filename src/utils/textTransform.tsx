export const upperCaseFirst = (str: String = '') => {
  return str.toLowerCase().replace(/^\w/, (chars) => chars.toUpperCase());
};
