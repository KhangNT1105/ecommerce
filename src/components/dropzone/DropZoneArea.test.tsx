import * as React from 'react';
import { shallow } from 'enzyme';

import { testSnapshots } from '../../utils/test';
import DropZoneArea from './DropZoneArea';

jest.mock('localforage');

const defaultProps = {
  apiURL: '',
  files: [],
  autoUpload: true,
  config: {
    accept: 'image/*',
    disabled: false,
    noClick: false,
    noKeyboard: false,
    minSize: 0,
    maxSize: 1048576 /* 1 MB */
  }
};

describe('<DropZoneArea />', () => {
  testSnapshots(DropZoneArea, [
    {
      props: { ...defaultProps },
      description: 'render snapshot correctly'
    }
  ]);
});

describe('Test logic', () => {
  test('dropzone', () => {
    const wrapper = shallow(<DropZoneArea {...defaultProps} />);
    // console.log(wrapper.find('.dropzone').props());
  });
});
