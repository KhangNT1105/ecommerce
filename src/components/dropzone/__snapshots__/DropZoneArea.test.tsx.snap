// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropZoneArea /> snapshots render snapshot correctly 1`] = `
<section
  className="dropzone--wrapper"
>
  <div
    className="dropzone"
    onBlur={[Function]}
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <div>
      <i
        className="fa fa-cloud-upload fa-fw fa-3x mb-3"
      />
      <form
        encType="multipart/form-data"
      >
        <input
          accept="image/*"
          autoComplete="off"
          multiple={true}
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          tabIndex={-1}
          type="file"
        />
      </form>
      <h5
        className="mt-0"
      >
        DROPZONE_MENTION_TEXT
      </h5>
      <div
        className="error-container"
      >
        <small
          className="small"
        >
          DROPZONE_IS_DRAG_ACTIVE
        </small>
        <small
          className="small"
        />
        <small
          className="small"
        />
        <small
          className="small"
        />
        <small
          className="small"
        >
          
        </small>
      </div>
    </div>
  </div>
</section>
`;
